src/android/MainActivity.java

```diff
@@ -0,0 +1,656 @@
+        // UI
+        isRecordingVideo = false;
+        myTimer.cancel();
+        try {
+            previewSession.stopRepeating();
+            previewSession.abortCaptures();
+        } catch (CameraAccessException e) {
+            e.printStackTrace();
+        }
+        // Stop recording
+        recorder.stop();
+        recorder.reset();
+
+        Intent intent = new Intent();
+        intent.putExtra("video_url", lastRecordedFileUrl);
+        setResult(RESULT_OK, intent);
+        finish();
+    }
+
+    private void updatePreview() {
+        if (null == cameraDevice) {
+            return;
+        }
+        try {
+            setUpCaptureRequestBuilder(previewBuilder);
+            HandlerThread thread = new HandlerThread("CameraPreview");
+            thread.start();
+            previewSession.setRepeatingRequest(previewBuilder.build(), null, backgroundHandler);
+        } catch (CameraAccessException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void setUpCaptureRequestBuilder(CaptureRequest.Builder builder) {
+        builder.set(CaptureRequest.CONTROL_MODE, CameraMetadata.CONTROL_MODE_AUTO);
+    }
+
+    public void openCamera(int width, int height) {
+        try {
+
+            try {
+                CameraManager manager = (CameraManager) getSystemService(Context.CAMERA_SERVICE);
+                String[] cameraIds = manager.getCameraIdList();
+
+                String camerId = manager.getCameraIdList()[0];
+                CameraCharacteristics characteristics = manager.getCameraCharacteristics(camerId);
+                StreamConfigurationMap map = characteristics.get(CameraCharacteristics.SCALER_STREAM_CONFIGURATION_MAP);
+
+                sensorOrientation = characteristics.get(CameraCharacteristics.SENSOR_ORIENTATION);
+                if (map == null) {
+                    throw new RuntimeException("Cannot get available preview/video sizes");
+                }
+                viewSizeCalculator.chooseVideoSize(map.getOutputSizes(MediaRecorder.class));
+                int orientation = getResources().getConfiguration().orientation;
+
+                previewsize = viewSizeCalculator.chooseOptimalSize(map.getOutputSizes(SurfaceTexture.class),
+                        width, height, orientation);
+                if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
+                    textureView.setAspectRatio(previewsize.getWidth(), previewsize.getHeight());
+                } else {
+                    textureView.setAspectRatio(previewsize.getHeight(), previewsize.getWidth());
+                }
+                configureTransform(width, height);
+                recorder = new MediaRecorder();
+
+                manager.openCamera(camerId, stateCallback, null);
+            } catch (CameraAccessException e) {
+                e.printStackTrace();
+            }
+        } catch (Exception e) {
+            Log.d("MyCameraApp", e.getMessage());
```

Comment:
All exceptions from openCamera are only logged, probably should throw error, maybe remove the try catch or add the throw?

Reply:
error are thrown